File: 3094-minimum-number-of-operations-to-make-array-empty/minimum-number-of-operations-to-make-array-empty.py
from collections import defaultdict
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        freq = defaultdict(int)
        for n in nums:
            freq[n]+=1
        result = 0
        for num, cnt in freq.items():
            if cnt == 1:
                return -1
            result += cnt//3
            if cnt%3 != 0:
                result += 1
        return result

File: 3094-minimum-number-of-operations-to-make-array-empty/minimum-number-of-operations-to-make-array-empty.go
func minOperations(nums []int) int {
    freq := map[int]int{}
    for _, n := range nums {
        _, found := freq[n]
        if (!found) {
            freq[n]=1
        } else {
            freq[n]=freq[n]+1
        }
    }
    result := 0
    for _, cnt := range freq {
        fmt.Println(cnt)
        if (cnt == 1) {
            return -1
        }
        result += cnt/3
        if (cnt%3!=0) {
            result += 1
        }
    }
    return result
}